File: 디자인패턴/java 디자인 패턴/PactoryMethod/Factory.java
abstract class Factory {
    protected abstract boolean isCreatable(String name);
    protected abstract Item createItem(String name);
    protected abstract void postProcessItem(String name);

    public Item create(String name) {
        boolean bCreateble = this.isCreatable(name);
        if(bCreateble) {
            Item item = this.createItem(name);
            postProcessItem(name);
            return item;
        }
        return null;
    }
}

File: 디자인패턴/Python 디자인 패턴/PactoryMethod/Factory.py
from abc import*
from Item import*
class Factory(ABC):
    @abstractmethod
    def isCreatable(this, name:str)->bool:pass
    @abstractmethod
    def CreateItem(this, name:str)->Item:pass
    @abstractmethod
    def postProcessItem(this, name:str):pass
    
    def create(this, name:str)->Item:
        if(this.isCreatable(name=name)):
            this.postProcessItem(name=name)
            return this.CreateItem(name=name)
        else:pass
    pass